From 2f2fde48594ec34e93ab409cd83442efe58e10ad Mon Sep 17 00:00:00 2001
From: Brendan Heading <brendanheading@gmail.com>
Date: Mon, 31 Aug 2015 15:24:44 +0100
Subject: [PATCH 3/3] fix build with musl

Add needed headers they are just not needed for glibc6+ but also
for musl
Define additional TCOPTS if not there

u_initX types are in sys/types.h be explicit about it

Upstream-Status: Pending

bh: this is a copy of the patch at the link below, modified to remove
the changes to include/libiptc/ipt_kernel_headers.h as these are
already integrated in the upstream tree. See :

http://lists.openembedded.org/pipermail/openembedded-core/2015-April/103613.html

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Brendan Heading <brendanheading@gmail.com>
---
 extensions/libxt_TCPOPTSTRIP.c           | 15 +++++++++++++++
 include/linux/netfilter_ipv4/ip_tables.h |  1 +
 2 files changed, 16 insertions(+)

diff --git a/extensions/libxt_TCPOPTSTRIP.c b/extensions/libxt_TCPOPTSTRIP.c
index 6897857..8a170b2 100644
--- a/extensions/libxt_TCPOPTSTRIP.c
+++ b/extensions/libxt_TCPOPTSTRIP.c
@@ -12,6 +12,21 @@
 #ifndef TCPOPT_MD5SIG
 #	define TCPOPT_MD5SIG 19
 #endif
+#ifndef TCPOPT_MAXSEG
+#	define TCPOPT_MAXSEG 2
+#endif
+#ifndef TCPOPT_WINDOW
+#	define TCPOPT_WINDOW 3
+#endif
+#ifndef TCPOPT_SACK_PERMITTED
+#	define TCPOPT_SACK_PERMITTED 4
+#endif
+#ifndef TCPOPT_SACK
+#	define TCPOPT_SACK 5
+#endif
+#ifndef TCPOPT_TIMESTAMP
+#	define TCPOPT_TIMESTAMP 8
+#endif
 
 enum {
 	O_STRIP_OPTION = 0,
diff --git a/include/linux/netfilter_ipv4/ip_tables.h b/include/linux/netfilter_ipv4/ip_tables.h
index 57fd82a..4807246 100644
--- a/include/linux/netfilter_ipv4/ip_tables.h
+++ b/include/linux/netfilter_ipv4/ip_tables.h
@@ -15,6 +15,7 @@
 #ifndef _IPTABLES_H
 #define _IPTABLES_H
 
+#include <sys/types.h>
 #include <linux/types.h>
 
 #include <linux/netfilter_ipv4.h>
-- 
2.4.3

